### PR TITLE
[#4120] fix(catalog-jdbc): Set partitioning when loading JDBC tables

### PR DESCRIPTION
**Pull request by xiaozcy**
### What changes were proposed in this pull request?

Set partitioning when loading JDBC tables.

### Why are the changes needed?

Fix: #4120 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

UT & IT.

### CODE REVIEW

**Comment by xiaozcy** [comment]
@mchades, could you help review this?

File: catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTableOperations.java [hunk 1]
```diff
@@ -308,6 +311,22 @@ protected List<Index> getIndexes(Connection connection, String databaseName, Str
...
+      Connection connection, String databaseName, String tableName) throws SQLException {
+    String showCreateTableSql = String.format("SHOW CREATE TABLE `%s`", tableName);
+    try (Statement statement = connection.createStatement();
+        ResultSet result = statement.executeQuery(showCreateTableSql)) {
+      StringBuilder createTableSql = new StringBuilder();
+      while (result.next()) {
+        createTableSql.append(result.getString("Create Table"));
+      }
+      Optional<Transform> transform =
+          DorisUtils.extractPartitionInfoFromSql(createTableSql.toString());
```
Comment (mchades) [inline catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTableOperations.java:325]: It seems more efficient to pass only the line containing the `PARTITION BY` keyword to `DorisUtils.extractPartitionInfoFromSql` instead of concatenating and splitting SQL lines.

File: catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/utils/DorisUtils.java [hunk 2]
```diff
@@ -65,4 +76,40 @@ public static Map<String, String> extractPropertiesFromSql(String createTableSql
     }
     return properties;
   }
+
+  public static Optional<Transform> extractPartitionInfoFromSql(String createTableSql) {
+    try {
+      String[] lines = createTableSql.split("\n");
+      for (String line : lines) {
+        if (line.contains(PARTITION_BY)) {
```
Comment (mchades) [inline catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/utils/DorisUtils.java:84]: Since the syntax of the partition by statement is fixed, we can use a regular here to make the code clear

File: integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/ITUtils.java [hunk 3]
```diff
@@ -85,6 +86,7 @@ public static void assertionsTableInfo(
       List<Column> columns,
       Map<String, String> properties,
       Index[] indexes,
+      Transform[] partitioning,
```
Comment (mchades) [inline integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/ITUtils.java:89]: we can add a new method to avoid some unnecessary changes of caller

File: catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTableOperations.java (hunk 1 above)
Comment (xiaozcy) [reply catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTableOperations.java:325]: We did not concatenate SQL here actually. When we execute `show create table` by JDBC, it will return the entire statement at once, which means the above loop will actually be executed once. So splitting SQL is necessary.
Besides, passing the entire statement to `DorisUtils.extractPartitionInfoFromSql` makes it easier for unit tests.

File: integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/ITUtils.java (hunk 3 above)
Comment (xiaozcy) [reply integration-test-common/src/test/java/com/datastrato/gravitino/integration/test/util/ITUtils.java:89]: Since we already set partitioning when loading tables, I think it's a necessary change in `assertionsTableInfo` to check it for all of the existing and future tests.

File: catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTableOperations.java (hunk 1 above)
Comment (mchades) [reply catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTableOperations.java:325]: If so, can we change above `WHILE` statement to `IF`?

File: catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/utils/DorisUtils.java (hunk 2 above)
Comment (xiaozcy) [reply catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/utils/DorisUtils.java]: done

File: catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTableOperations.java (hunk 1 above)
Comment (xiaozcy) [reply catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTableOperations.java:325]: done

**Review by mchades — APPROVED** [review]
LGTM, thanks for your contributions!
